File: src/perceptron/Perceptron.java
package perceptron;

import java.util.List;

public class Perceptron implements IPerceptron {
	private final List<Integer> weights;
	private final int bias;

	protected Perceptron(final List<Integer> weights, final int bias) {
		this.weights = weights;
		this.bias = bias;
	}

	@Override
	public boolean eval(final List<Integer> inputs) {
		int sum = 0;
		for(int i=0; i < weights.size() ; i++) {
			sum += inputs.get(i)*weights.get(i);
		}
		return sum + bias>= 0;
	}

}

File: Class 02/perceptron/Perceptron.py
import numpy as np

learningRate = 0.01

class Perceptron:

    def __init__(self, weightList, bias):
        self.weights = np.array(weightList)
        self.bias = bias

    def evaluate(self, inputList):
        try:
            assert len(self.weights) == len(inputList)

            resSum = sum(self.weights * np.array(inputList))
            res = 0
            if resSum + self.bias > 0:
                res = 1
            return res

        except AssertionError:
            print("The length of the input defers from the weights vector.")
        except TypeError:
            print("You either entered a non number list in constructor or input. Types don't match.")

    def train(self, trainInputList, expectedResult):
        localResult = self.evaluate(trainInputList)
        if localResult > expectedResult:
            self.decreaseWeights(trainInputList)
        elif localResult < expectedResult:
            self.increaseWeights(trainInputList)


    def decreaseWeights(self, inputs):
        self.weights -= np.array(inputs) * learningRate

    def increaseWeights(self, inputs):
        self.weights += np.array(inputs) * learningRate
